### PR TITLE
[pythonic config] Allow using 'resource_defs' with resource args in assets

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Fixes small bug discovered in dogfooding today w/ #12664. Ensures that asset decorators can be passed both `resource_defs` and specify a list of resources as parameters:

```python
@asset(
    resource_defs = {"my_resource": MyResource(...)}
)
def my_asset(my_resource: MyResource):
    pass
```

## Test Plan

New unit test.

### CODE REVIEW

**Comment by benpankow** [comment]
Current dependencies on/for this PR:
* master
  * **PR #12679** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12679" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12679?utm_source=stack-comment).

File: python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py [hunk 1]
```diff
@@ -306,14 +308,17 @@ def __call__(self, fn: Callable) -> AssetsDefinition:
                 code_version=self.code_version,
             )
 
+            op_required_resource_keys = decorator_resource_keys - arg_resource_keys
+            print("op_required_resource_keys", op_required_resource_keys)
```
Comment (schrockn) [inline python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py:312]: rm

File: python_modules/dagster/dagster_tests/core_tests/resource_tests/test_struct_config_resources.py [hunk 2]
```diff
@@ -1229,3 +1229,29 @@ def an_asset(a_resource: OuterResource):
...
+
+    defs = Definitions(
+        assets=[an_asset],
+    )
+    defs.get_implicit_global_asset_job_def().execute_in_process()
+
+    assert executed["yes"]
+
+    # Cannot specify both required_resource_keys and resources as args
+    with pytest.raises(Exception):
```
Comment (schrockn) [inline python_modules/dagster/dagster_tests/core_tests/resource_tests/test_struct_config_resources.py:1253]: nit: catch more specific exception (`CheckError`?)

**Review by schrockn — APPROVED** [review]
please heed final comments!

**Pull request merged**